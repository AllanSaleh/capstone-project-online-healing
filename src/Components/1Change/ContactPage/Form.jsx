import React from 'react';

export default function Form() {
  return (
    <div className="flex flex-col px-sides sm:mt-0 mt-20 h-section items-center sm:flex-row w-full">
      <form className="flex flex-col sm:w-1/2 w-full">
        <div className="mb-8">
          <label className="mb-2  block text-paragraph">Full Name:</label>
          <input
            className="p-2 w-3/4 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
            placeholder="Enter your full name"
          />
        </div>
        <div className="mb-8">
          <label className="mb-2 block text-paragraph">Email:</label>
          <input
            className="p-2 w-3/4 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
            placeholder="Enter your email address"
          />
        </div>
        <div className="mb-8">
          <label className="mb-2 block text-paragraph">Details:</label>
          <textarea
            className="p-2 w-3/4 h-52 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
            placeholder="Enter your details here..."
          />
        </div>
        <button
          className="w-32 h-12 text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div className="sm:w-1/2 w-full flex justify-center ">
        <div className="bg-blue-light rounded-lg p-6 sm:w-72 w-56 h-64  ">
          <div className="text-subtitle mb-4">Find Us At:</div>
          <div className="text-paragraph text-gray-700">Nergiz Plaza</div>
          <div className="text-paragraph text-gray-700">3rd Floor</div>
          <div className="text-paragraph text-gray-700">Bakhtiyari Street 40m</div>
          <div className="text-paragraph text-gray-700">Erbil, Iraq</div>
          <div className="text-paragraph text-gray-700">44001</div>
        </div>
      </div>
    </div>
  );
}
