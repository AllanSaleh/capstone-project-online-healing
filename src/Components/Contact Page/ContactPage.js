import React from 'react';

import NavBar from '../NavBar/NavBar';
import FirstSection from './FirstSection';
import SecondSection from './SecondSection';
import Footer from '../Footer/Footer';


export default function ContactPage() {
  return (
    <div>
      <NavBar />
      <FirstSection />
      <SecondSection />
      <Footer />
    </div>
  );
}
