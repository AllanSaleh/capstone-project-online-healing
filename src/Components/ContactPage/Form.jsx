import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';

export default function Form({ selectedChoice }) {
  const history = useHistory();

  const [inputs, setInputs] = useState({
    name: '',
    email: '',
    details: '',
  });

  const GoToRoute = () => {
    const verify = [false, false, false, false];

    if (selectedChoice !== '') verify[3] = true;
    else {
      verify[3] = false;
      alert('Please choose a type of contact!');
    }
    if (inputs.name.match(/[a-zA-Z]/)) verify[0] = true;
    else {
      verify[0] = false;
      alert('Incorrect Name Format');
    }
    if (inputs.email.match(/[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9]+[.]+[a-zA-Z0-9]/))
      verify[1] = true;
    else {
      verify[1] = false;
      alert('Incorrect E-Mail Format');
    }
    if (inputs.details !== '') verify[2] = true;
    else {
      verify[2] = false;
      alert('Please provide details!');
    }

    let AllVerified = true;
    for (let i = 0; i < verify.length; i += 1) if (!verify[i]) AllVerified = false;
    if (AllVerified) {
      window.scrollTo(0, 0);
      history.push({
        pathname: '/ThankYou',
        state:
          'Your request has been sent, a member of the support team will get in contact with you through the email you provided as soon as possible.',
      });
    }
  };

  return (
    <div className="flex flex-col px-sides sm:mt-0 mt-20 h-section items-center sm:flex-row w-full">
      <form className="flex flex-col sm:w-1/2 w-full">
        <div className="mb-8">
          <label className="mb-2  block text-paragraph">Full Name:</label>
          <input
            onBlur={(e) => setInputs({ ...inputs, name: e.target.value })}
            className="p-2 w-3/4 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
            placeholder="Enter your full name"
          />
        </div>
        <div className="mb-8">
          <label className="mb-2 block text-paragraph">Email:</label>
          <input
            onBlur={(e) => setInputs({ ...inputs, email: e.target.value })}
            className="p-2 w-3/4 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
            placeholder="Enter your email address"
          />
        </div>
        <div className="mb-8">
          <label className="mb-2 block text-paragraph">Details:</label>
          <textarea
            onBlur={(e) => setInputs({ ...inputs, details: e.target.value })}
            className="p-2 w-3/4 h-52 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
            placeholder="Enter your details here..."
          />
        </div>
        <button
          onClick={() => GoToRoute()}
          className="w-32 h-12 text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
          type="button"
        >
          Submit
        </button>
      </form>
      <div className="sm:w-1/2 w-full flex justify-center ">
        <div className="bg-blue-light rounded-lg p-6 sm:w-72 w-56 h-64  ">
          <div className="text-subtitle mb-4">Find Us At:</div>
          <div className="text-paragraph text-gray-700">Nergiz Plaza</div>
          <div className="text-paragraph text-gray-700">3rd Floor</div>
          <div className="text-paragraph text-gray-700">Bakhtiyari Street 40m</div>
          <div className="text-paragraph text-gray-700">Erbil, Iraq</div>
          <div className="text-paragraph text-gray-700">44001</div>
        </div>
      </div>
    </div>
  );
}
