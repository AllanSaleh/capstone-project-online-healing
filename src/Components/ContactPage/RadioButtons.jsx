import React from 'react';

import ContactPic from './Images/ContactPic.svg';

export default function RadioButtons() {
  const radios = [
    { value: 'ServiceQuestion', text: 'I have a question about the service.' },
    {
      value: 'ClientSupport',
      text: "I'm a registered client and I need support.",
    },
    {
      value: 'CouncelorInterest',
      text: "I'm a counselor interested in joining.",
    },
    {
      value: 'CounselorSupport',
      text: "I'm a counselor interested in joining.",
    },
    {
      value: 'BusinessInquiry',
      text: "I'm a registered counselor and I need support.",
    },
    {
      value: 'OrganizationInterest',
      text: "I'm interested in Healing Online for my organization.",
    },
    { value: 'BillingQuestion', text: 'I have a billing related question.' },
  ];
  return (
    <div className="h-firstsection pt-navbar px-sides flex flex-col justify-evenly w-full">
      <div>
        <div className="text-2xl sm:text-title sm:mb-0 mb-3">
          SEND US YOUR REQUEST!
        </div>
        <div className="sm:text-subtitle sm:mb-0 mb-10  text-gray-700 md:text-sm lg:text-xl lg:text-gray-700 lg:text-subtitle md:mt-5 md:flex-col md:justify-evenly md:flex">
          Do you have a question, concern, idea, feedback, or problem? If you
          need assistance, please fill out the form below and we'd be happy to
          help!
        </div>
      </div>

      <div className="flex justify-between sm:flex-row flex-col">
        <div>
          <div className="text-subtitle mb-4 text-sm sm:text-lg">Type of Contact</div>
          <div className="flex flex-col justify-between">
            {radios.map((radio) => (
              <div className="flex ml-4 my-2 ">
                <input
                  name="ContactRadio"
                  className="my-2 mr-4"
                  type="radio"
                  value={radio.vlaue}
                />
                <div className="text-paragraph sm:text-lg text-xm">{radio.text}</div>
              </div>
            ))}
          </div>
        </div>
        <img src={ContactPic} className="sm:block hidden" alt="contact pic" />
      </div>
    </div>
  );
}
