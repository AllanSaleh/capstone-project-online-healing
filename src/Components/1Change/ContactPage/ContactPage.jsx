import React from 'react';

import RadioButtons from './RadioButtons';
import Form from './Form';

export default function ContactPage() {
  return (
    <>
      <RadioButtons />
      <Form />
    </>
  );
}
