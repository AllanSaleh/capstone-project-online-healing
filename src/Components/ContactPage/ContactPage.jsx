import React, { useState } from 'react';

import RadioButtons from './RadioButtons';
import Form from './Form';

export default function ContactPage({ loginStatus }) {
  window.scrollTo(0, 0);
  const [selectedChoice, setChoice] = useState('');

  return (
    <>
      <RadioButtons
        choiceCheck={(choice) => {
          if (choice !== selectedChoice) setChoice(choice);
        }}
      />
      <Form selectedChoice={selectedChoice} loginStatus={loginStatus} />
    </>
  );
}
