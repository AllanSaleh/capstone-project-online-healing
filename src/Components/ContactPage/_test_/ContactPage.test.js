import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ContactPage from '../ContactPage';

it(' Contact Page renders correctly', () => {
  const tree = renderer.create(<ContactPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
