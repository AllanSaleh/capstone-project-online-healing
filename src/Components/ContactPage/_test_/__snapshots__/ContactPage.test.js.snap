// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="h-firstsection pt-navbar px-sides flex flex-col justify-evenly w-full"
  >
    <div>
      <div
        className="text-2xl sm:text-title sm:mb-0 mb-3"
      >
        SEND US YOUR REQUEST!
      </div>
      <div
        className="sm:text-subtitle sm:mb-0 mb-10  text-gray-700 md:text-sm lg:text-xl lg:text-gray-700 lg:text-subtitle md:mt-5 md:flex-col md:justify-evenly md:flex"
      >
        Do you have a question, concern, idea, feedback, or problem? If you need assistance, please fill out the form below and we'd be happy to help!
      </div>
    </div>
    <div
      className="flex justify-between sm:flex-row flex-col"
    >
      <div>
        <div
          className="text-subtitle mb-4 text-sm sm:text-lg"
        >
          Type of Contact
        </div>
        <div
          className="flex flex-col justify-between"
        >
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I have a question about the service.
            </div>
          </div>
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I'm a registered client and I need support.
            </div>
          </div>
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I'm a counselor interested in joining.
            </div>
          </div>
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I'm a counselor interested in joining.
            </div>
          </div>
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I'm a registered counselor and I need support.
            </div>
          </div>
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I'm interested in Healing Online for my organization.
            </div>
          </div>
          <div
            className="flex ml-4 my-2 "
          >
            <input
              className="my-2 mr-4"
              name="ContactRadio"
              type="radio"
            />
            <div
              className="text-paragraph sm:text-lg text-xm"
            >
              I have a billing related question.
            </div>
          </div>
        </div>
      </div>
      <img
        alt="contact pic"
        className="sm:block hidden"
        src="ContactPic.svg"
      />
    </div>
  </div>,
  <div
    className="flex flex-col px-sides sm:mt-0 mt-20 h-section items-center sm:flex-row w-full"
  >
    <form
      className="flex flex-col sm:w-1/2 w-full"
    >
      <div
        className="mb-8"
      >
        <label
          className="mb-2  block text-paragraph"
        >
          Full Name:
        </label>
        <input
          className="p-2 w-3/4 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
          placeholder="Enter your full name"
        />
      </div>
      <div
        className="mb-8"
      >
        <label
          className="mb-2 block text-paragraph"
        >
          Email:
        </label>
        <input
          className="p-2 w-3/4 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
          placeholder="Enter your email address"
        />
      </div>
      <div
        className="mb-8"
      >
        <label
          className="mb-2 block text-paragraph"
        >
          Details:
        </label>
        <textarea
          className="p-2 w-3/4 h-52 text-paragraph text-gray-700 rounded-lg border-2 border-gray-400 shadow-lg"
          placeholder="Enter your details here..."
        />
      </div>
      <button
        className="w-32 h-12 text-subtitle bg-blue-dark rounded-lg border-2 border-transparent hover:bg-white hover:text-blue-dark hover:border-blue-dark"
        type="submit"
      >
        Submit
      </button>
    </form>
    <div
      className="sm:w-1/2 w-full flex justify-center "
    >
      <div
        className="bg-blue-light rounded-lg p-6 sm:w-72 w-56 h-64  "
      >
        <div
          className="text-subtitle mb-4"
        >
          Find Us At:
        </div>
        <div
          className="text-paragraph text-gray-700"
        >
          Nergiz Plaza
        </div>
        <div
          className="text-paragraph text-gray-700"
        >
          3rd Floor
        </div>
        <div
          className="text-paragraph text-gray-700"
        >
          Bakhtiyari Street 40m
        </div>
        <div
          className="text-paragraph text-gray-700"
        >
          Erbil, Iraq
        </div>
        <div
          className="text-paragraph text-gray-700"
        >
          44001
        </div>
      </div>
    </div>
  </div>,
]
`;
